pymc/logprob/abstract.py

```diff
@@ -131,14 +131,36 @@ def _icdf_helper(rv, value, **kwargs):
     return rv_icdf
 
 
+from enum import Enum, auto
+
+
+class MeasureType(Enum):
+    Discrete = auto()
+    Continuous = auto()
+    Mixed = auto()
+
+
 class MeasurableVariable(abc.ABC):
     """A variable that can be assigned a measure/log-probability"""
 
+    def __init__(
+        self,
+        *args,
+        ndim_supp: Union[int, Tuple[int]],
+        supp_axes: Tuple[Union[int, Tuple[int]]],
+        measure_type: Union[MeasureType, Tuple[MeasureType]],
```

Comment:
Should not that be a frozenset or some collection? Union requires if else later, does not it?

Reply:
Can you explain what you mean in more detail?